### PR TITLE
fix for #327, large reduction in recipe variables written to env.data.pt DB

### PR DESCRIPTION
**Pull request by rbaynes**

### CODE REVIEW

File: nodes/sensor_persistence.py [hunk 1]
```diff
@@ -18,13 +17,14 @@
 from openag_lib.db_bootstrap.db_names import ENVIRONMENTAL_DATA_POINT
 from openag_lib.config import config as cli_config
 from openag_brain.models import EnvironmentalDataPoint
-from openag_brain.utils import read_environment_from_ns
+from openag_brain.utils import gen_doc_id, read_environment_from_ns
```
Comment (rbaynes) [inline nodes/sensor_persistence.py]: use the same doc ID creating function as everyone else!

File: nodes/recipe_persistence.py [hunk 2]
```diff
@@ -26,46 +27,74 @@
...
+        self.environment = environment
+        self.variable = variable
+        self.is_desired = is_desired
+        self.last_value = None
+        self.sub = rospy.Subscriber(topic, topic_type, self.on_data)
+
+    def on_data(self, item):
+        curr_time = time.time()
+        value = item.data
+        if value is None or value == self.last_value:
```
Comment (rbaynes) [inline nodes/recipe_persistence.py]: don't keep writing the same value!

File: nodes/sensor_persistence.py [hunk 3]
```diff
@@ -65,15 +65,14 @@ def on_data(self, item):
...
+        point_id = gen_doc_id(curr_time)
         self.db[point_id] = point
 
         self.last_value = value
         self.last_time = curr_time
 
-    def gen_doc_id(self, curr_time):
-        return "{}-{}".format(curr_time, random.randint(0, sys.maxsize))
 
+#-----------------------------------------------------------------------------
```
Comment (Spaghet) [inline nodes/sensor_persistence.py]: 👎

File: nodes/sensor_persistence.py [hunk 4]
```diff
@@ -89,6 +88,8 @@ def create_persistence_objects(
             min_update_interval=min_update_interval
         )
 
+
+#-----------------------------------------------------------------------------
```
Comment (Spaghet) [inline nodes/sensor_persistence.py]: 👎

File: nodes/sensor_persistence.py [hunk 5]
```diff
@@ -18,13 +17,14 @@
...
 from openag_brain.models import EnvironmentalDataPoint
-from openag_brain.utils import read_environment_from_ns
+from openag_brain.utils import gen_doc_id, read_environment_from_ns
 from openag_brain.load_env_var_types import create_variables
 
 # Filter a list of environmental variables that are specific to environment
 # sensors and actuators
 ENVIRONMENT_VARIABLES = create_variables(rospy.get_param('/var_types/environment_variables'))
 
+#-----------------------------------------------------------------------------
```
Comment (Spaghet) [inline nodes/sensor_persistence.py]: 👎

File: nodes/sensor_persistence.py (hunk 1 above)
Comment (Spaghet) [inline nodes/sensor_persistence.py]: nice find 👍

File: nodes/recipe_persistence.py [hunk 6]
```diff
@@ -26,46 +27,74 @@
 VALID_VARIABLES = ENVIRONMENTAL_VARIABLES.union(RECIPE_VARIABLES)
 
 
+#-----------------------------------------------------------------------------
+class TopicPersistence:
```
Comment (Spaghet) [inline nodes/recipe_persistence.py]: classes 😭 😞 😢

File: nodes/recipe_persistence.py [hunk 7]
```diff
@@ -26,46 +27,74 @@
...
+            "is_desired": self.is_desired,
+            "value": value,
+            "timestamp": curr_time
+        })
+        point_id = gen_doc_id(curr_time)
+        self.db[point_id] = point
+        self.last_value = value
+
+
+#-----------------------------------------------------------------------------
```
Comment (Spaghet) [inline nodes/recipe_persistence.py]: 👎

File: nodes/recipe_persistence.py [hunk 8]
```diff
@@ -26,46 +27,74 @@
...
+        TopicPersistence(
+            db=env_var_db, 
+            topic=topic, 
+            topic_type=get_message_class(variable.type),
+            environment=environment_id,
+            variable=variable.name, is_desired=True
+        )
+
+
+#-----------------------------------------------------------------------------
```
Comment (Spaghet) [inline nodes/recipe_persistence.py]: 👎

**Comment by rbaynes** [comment]
Dude, give some old eyes a break :)   I'll remove them if they bug you that
much.


Thanks,
Rob Baynes

**Pull request merged**